parsing/unit_info.mli

```diff
@@ -94,6 +106,8 @@ val make:
     ?check_modname:bool -> source_file:filename ->
     intf_or_impl -> file_prefix -> t
 
+val set_original_source_file_name : t -> filename -> t
```

Comment:
`set` tends to indicate a mutable action. I would propose `update_original_source_file_name`.